plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeModernizerSession.kt

```diff
@@ -342,24 +277,55 @@ class CodeModernizerSession(
                 state.currentJobStatus = new
                 state.transformationPlan = plan
                 sessionContext.project.refreshCwQTree()
-                if (new == TransformationStatus.PARTIALLY_COMPLETED) {
-                    isPartialSuccess = true
+                // Open the transformation plan detail panel once transformation plan is available
+                if (state.transformationPlan != null && !isTransformationPlanEditorOpened) {
+                    tryOpenTransformationPlanEditor()
+                    isTransformationPlanEditorOpened = true
                 }
                 val instant = Instant.now()
+                // Set the job start time
+                if (state.currentJobCreationTime == Instant.MIN) {
+                    state.currentJobCreationTime = instant
+                }
                 state.updateJobHistory(sessionContext, new, instant)
                 setCurrentJobStopTime(new, instant)
+                setCurrentJobSummary(new)
+
+                if (!passedStart && new in STATES_AFTER_STARTED) {
+                    passedStart = true
+                }
+                if (!passedBuild && new in STATES_AFTER_INITIAL_BUILD) {
+                    passedBuild = true
+                }
+
                 jobTransitionHandler(new, plan)
                 LOG.info { "Waiting for Modernization Job [$jobId] to complete. State changed for job: $old -> $new" }
             }
             return when {
                 result.state == TransformationStatus.STOPPED -> CodeModernizerJobCompletedResult.Stopped
-                isPartialSuccess -> CodeModernizerJobCompletedResult.JobPartiallySucceeded(jobId, sessionContext.targetJavaVersion)
-                result.succeeded -> CodeModernizerJobCompletedResult.JobCompletedSuccessfully(jobId)
                 result.state == TransformationStatus.UNKNOWN_TO_SDK_VERSION -> CodeModernizerJobCompletedResult.JobFailed(
                     jobId,
                     message("codemodernizer.notification.warn.unknown_status_response")
                 )
 
+                result.state == TransformationStatus.PARTIALLY_COMPLETED -> CodeModernizerJobCompletedResult.JobPartiallySucceeded(jobId, sessionContext.targetJavaVersion)
+
+                result.state == TransformationStatus.FAILED -> {
```

Comment:
Were we not handling this case before?

Reply:
They are handled in the existing code, but in `awaitModernizationPlan` function. These if statements determines what fallback error message to show based if we passed job start or job pre-build.